src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedFieldSymbol.vb

```diff
@@ -59,6 +59,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Return
             End If
 
+            Dim compilation = Me.DeclaringCompilation
+            If Type.ContainsTupleNames() AndAlso
+                compilation.HasTupleNamesAttributes AndAlso
```

Comment:
Good question. The tuple names attribute actually should not be optional for things that are no synthetic. It should be a compile error if you use names and attribute is not avaialable (I missed this part)
The check inside SynthesizeTupleNamesAttributeOpt should always succeed.


---
In reply to: [79676084](https://github.com/dotnet/roslyn/pull/13923#discussion_r79676084) [](ancestors = 79676084)